forum/qa-plugin/report-reason/q2apro-flag-reasons-page.php

```diff
@@ -0,0 +1,209 @@
+                $processingFlagReasonError = call_user_func_array([$this, 'processFlagToQuestion'], $flagParams);
+                break;
+            }
+            case 'a': {
+                $processingFlagReasonError = call_user_func_array([$this, 'processFlagToAnswer'], $flagParams);
+                break;
+            }
+            case 'c': {
+                $processingFlagReasonError = call_user_func_array([$this, 'processFlagToComment'], $flagParams);
+                break;
+            }
+            default: {
+                $processingFlagReasonError = 'UNRECOGNIZED_POST_TYPE:' . $postType;
+            }
+        }
+
+        return $processingFlagReasonError;
+    }
+
+    private function processFlagToQuestion($userId, $postId, $questionId, $reasonId, $notice) {
+        list($question, $childPosts, $aChildPosts, $closePost, $duplicatePosts) = qa_db_select_with_pending(
+            qa_db_full_post_selectspec($userId, $questionId),
+            qa_db_full_child_posts_selectspec($userId, $questionId),
+            qa_db_full_a_child_posts_selectspec($userId, $questionId),
+            qa_db_post_parent_q_selectspec($questionId),
+            false
+        );
+
+        $questionFlagError = qa_flag_error_html($question, $userId, $questionId);
+        if ($questionFlagError) {
+            return $questionFlagError;
+        }
+
+        $answers = qa_page_q_load_as($question, $childPosts);
+        $commentsFollows = qa_page_q_load_c_follows($question, $childPosts, $aChildPosts, $duplicatePosts);
+
+        if (qa_flag_set_tohide($question, $userId, qa_userid_to_handle($userId), qa_cookie_get(), $question)) {
+            qa_question_set_status(
+                $question, QA_POST_STATUS_HIDDEN, null, null, null, $answers, $commentsFollows, $closePost
+            ); // hiding not really by this user so pass nulls
+        }
+
+//        var_dump('<br>$reasonId <= $this->CUSTOM_REPORT_REASON_ID: ', $reasonId <= $this->CUSTOM_REPORT_REASON_ID, ' /$this->CUSTOM_REPORT_REASON_ID: ', $this->CUSTOM_REPORT_REASON_ID);
+        if($reasonId >= 0 && $reasonId <= $this->CUSTOM_REPORT_REASON_ID) {
+            qa_db_query_sub(
+            '
+                INSERT INTO `^flagreasons` (`userid`, `postid`, `reasonid`, `notice`)
+                VALUES (#, #, #, $)
+            ', $userId, $postId, $reasonId, $notice
+            );
+        } else {
+             $this->handleReportErrorAndExit('INVALID_REASON_ID');
+         }
+    }
+
+    private function processFlagToAnswer($userId, $postId, $questionId, $reasonId, $notice) {
+        list($answer, $question, $qChildPosts, $aChildPosts) = qa_db_select_with_pending(
+            qa_db_full_post_selectspec($userId, $postId),
+            qa_db_full_post_selectspec($userId, $questionId),
+            qa_db_full_child_posts_selectspec($userId, $questionId),
+            qa_db_full_child_posts_selectspec($userId, $postId)
+        );
+
+        // TODO: might not be needed
```

Comment:
to chyba też do usunięcia skoro niepotrzebne i działa albo jakiegoś wykorzystania, bo teraz ja np. nie wiem o co chodzi i tym bardziej za np. rok może być komukolwiek cięzko sobie przypomnieć co to

Reply:
Tak, docelowo to będzie usunięte.